Adds map overlay switch on map

TheYakuzo opened the pull request:
Introduces a button to toggle the overlay's visibility. The overlay preference persists across sessions.

Memoize MapOverlay to avoid unnecessary re-renders

Move activeOverlay management from MapOverlay.js to MainMap.jsx to avoid code duplication on MapOverlay.js and MapOverlayButton.js [ new component ]

The button is not rendered when no map overlay is indicated (hence the displacement of the activeOverlay logic).

![Screenshot 2025-04-05 at 18 34 07](https://github.com/user-attachments/assets/236cc90c-0f87-4a08-90fd-4fdf4115baf5)

![Screenshot 2025-04-05 at 18 34 17](https://github.com/user-attachments/assets/99df7d07-fbb6-40a3-b4b3-1dfd2fd707a8)

TheYakuzo committed:
Adds map overlay functionality

Introduces a button to toggle the overlay's visibility. The overlay preference persists across sessions.

Memoize MapOverlay to avoid unnecessary re-renders

TheYakuzo committed:
lint fix and refactors map overlay implementation.

Refactors the map overlay component to improve code readability and maintainability.

- Updates the overlay control to use static getDefaultPosition method.
- Updates the overlay styling to use single quotes.

tananaev commented on src/map/overlay/MapOverlayButton.js:
```diff
@@ -0,0 +1,53 @@
+
+const statusClass = (status) => `maplibregl-ctrl-icon maplibre-ctrl-overlay maplibre-ctrl-overlay-${status}`;
+
+class OverlayControl {
```
Is this basically a copy of the alarm button? I think we need to figure out how to reuse a button instead of copying.

TheYakuzo replied:
Yes, it's true, I'll see tomorrow to redo these two buttons.
Do you usually put them in src/common/components?

tananaev replied:
Map components should be under map folder.

TheYakuzo committed:
Replaces notification and overlay with panel buttons

Refactors the map notification and overlay buttons into a generic panel button component.

This change improves code reusability and simplifies the map interface by using a consistent button style for different map controls.

TheYakuzo committed:
Refactors map control button styling

Simplifies map control button styling by removing the redundant `maplibre-ctrl-background-icon` class.

This change improves code readability and maintainability by consolidating the icon styling under a single `maplibre-ctrl-icon` class, making it easier to manage button appearance.

tananaev commented:
Can we split into 2 PRs:
1. Refactor button to be generic
2. Add overlay control button

Also I think the icon shouldn't have a color.

TheYakuzo commented:
Done : https://github.com/traccar/traccar-web/pull/1358 ( refactoring )
This current PR remains the same, there should be no conflicts.

TheYakuzo committed:
Updates overlay icon colors

Updates the overlay icon colors to improve visual clarity.

The "on" state icon is now black, and the "off" state icon is now gray.

TheYakuzo commented:
color is now black for enabled and gray for disabled
